### PR TITLE
Megan Checks for Understanding Week 3

### PR DESCRIPTION
**Pull request by meganft**

### CODE REVIEW

**Comment by meganft** [comment]
This went pretty well! I am not totally solid on the _path vs _url since I have never used the _url.   I'm sure there are also some things I need to learn about forms and the strong params. Overall though, I feel pretty good about my understanding of these topics.

File: module-2/backend/week_three.md [hunk 1]
```diff
@@ -1,25 +1,56 @@
...
+`rails new project-name` 
+
+2. What do Models generally inherit from in rails?<br>
+`ActiveRecord::Base`
+
+3. What do Controllers generally inherit from in a rails project?<br>
+`ApplicationController`
+
+4. How would I create a route if I wanted to see a specific horse in my routes file assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?<br>
+`resources :horses, only: [:show]`
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: :+1:

File: module-2/backend/week_three.md [hunk 2]
```diff
@@ -1,25 +1,56 @@
...
+
+5. What rake task is useful when looking at routes, and what information does it give you?<br>
+`rake routes`. It shows you all current routes available in your rails app.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: Yup! To add a little more detail it shows you prefixes, verbs, URI patterns, and controller actions.

File: module-2/backend/week_three.md [hunk 3]
```diff
@@ -1,25 +1,56 @@
...
+`_url` is an absolute path rather than the relative path we have defined in our application (where we use `_path`). So if we are sending an email or linking to something ourside our app, we use _url.
+
+8. What are strong params and why are the necessary? <br>
+Strong params provide us with another level of security from submitted data. We only want to receive the permitted info so we define that in our controller.
+
+9. What role does `form_for` play in helping us create our forms? <br>
+It is a rails helper method that caputeres user input data.
+
+10. How does `form_for` know where to submit the user's input? <br>
+Based on the variable we passed to it. If it was a form_for @horse, it would know that it was dealing with an instance of a horse.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: Yup! It also checks to see if that instance has already been saved to the database to see where it should be submitting the form and if it should be a POST or PATCH request.

File: module-2/backend/week_three.md [hunk 4]
```diff
@@ -1,25 +1,56 @@
...
+  
+  <%= f.label age %>
+  <%= f.text_field age %>
+  
+  <%= f.submit %>
+<% end %>
+```
+
+12. Why do we want to validate our models?<br>
+We do this to ensure that the relationships between our tables works correctly.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: Maybe? I'd be interested to hear more. When I think of validations I think primarily about ensuring that all of the required information has been entered before we attempt to save to the database and that that information meets certain other requirements we might have for it.

File: module-2/backend/week_three.md [hunk 5]
```diff
@@ -1,25 +1,56 @@
...
+`ApplicationController`
+
+4. How would I create a route if I wanted to see a specific horse in my routes file assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?<br>
+`resources :horses, only: [:show]`
+
+5. What rake task is useful when looking at routes, and what information does it give you?<br>
+`rake routes`. It shows you all current routes available in your rails app.
+
+6. What is an example of a route helper? When would you use them?<br>
+A route helper is something like `edit_company_path(company)`. You would use this in your rails app when linking to the page to edit a specific company, in this instance.
```
Comment (s-espinosa) [inline module-2/backend/week_three.md]: :tada:

**Pull request closed without merging**